openmdao/core/system.py

```diff
@@ -1203,6 +1203,31 @@ def _list_conns(self, relname):
         else:
             _list_conns(self, '')
 
+    def list_states(self, stream=sys.stdout):
+        """
+        Recursively list all states and their initial values.
+
+        Args
+        ----
+        stream : output stream, optional
+            Stream to write the connection info to. Default is sys.stdout.
+        """
+
+        for uname, meta in iteritems(self._unknowns_dict):
+            states = []
+            if meta.get('state'):
+                states.append(uname)
+
+        if states:
+            stream.write("\nStates in model:\n")
```

Comment:
Should this say 'States in model:" when this can be called at any level and not just at the top?  Maybe replace model with the pathname of the system you call it on, or use 'model' if the pathname is empty?